pallets/creditcoin/src/lib.rs

```diff
@@ -558,11 +569,73 @@ pub mod pallet {
 	#[pallet::hooks]
 	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
 		fn on_initialize(block_number: T::BlockNumber) -> Weight {
-			log::debug!("Cleaning up expired entries");
+			log::debug!("Cleaning up expired entries: {block_number:?}");
+
+			let mut current =
+				CleanupState::<T>::get().unwrap_or_else(|| StorageCleanupState::new(block_number));
+
+			let limit = T::PerBlockCleanupLimit::get();
+			let mut ask_count = 0u32;
+			let mut bid_count = 0u32;
+			let mut offer_count = 0u32;
+			let mut total_count = 0u32;
+
+			loop {
+				let StorageCleanupState { ask_orders, bid_orders, offers } = current;
+				log::debug!("ask orders block: {:?}, cursor: {}; bid orders block: {:?}, cursor: {}; offers block: {:?}, cursor: {}",
+					ask_orders.on_block,ask_orders.cursor.is_some(), bid_orders.on_block, bid_orders.cursor.is_some(),offers.on_block, offers.cursor.is_some(),);
+				let ask_cleanup = (ask_orders.on_block <= block_number).then(|| {
```

Comment:
I would add ` && total_count < limit` here as well in case these blocks ever get reordered or copy pasted. 

My preference is that conditions are the same everywhere (even when not needed) to avoid silly mistakes from shifting code around.

Reply:
On a second though we may be going through a subsequent iteration and have more AskOrders to delete so having the condition makes sense.